DOC: fix doc build failing with pydata-sphinx-theme 0.10.1

`make html` fails with below error when using **pydata-sphinx-theme** 0.10.1

```
Theme error:
An error happened in rendering the page about.
Reason: KeyError('self')
make: *** [html] Error 2
```

re-create in a highly cut down env
```
import os, shutil
from pathlib import Path
from sphinx.cmd.build import build_main

a = [
    "-T",
    "-b",
    "html",
    "-d",
    "build/doctrees",
    "source",
    "build/html",
]

assert Path.cwd().name == "geopandas"
p = Path.cwd().parent.joinpath("simple_sphinx")
g = Path.cwd().joinpath("doc")
if p.exists():
    shutil.rmtree(p)

if not p.exists():
    p.mkdir()
    p2 = p.joinpath("doc")
    p2.mkdir()
    os.chdir(p2)
    os.system("sphinx-quickstart -p sphinx-text -a Rob -l en -v 0.0.1 -r 0.0.1 --sep")
p = p.joinpath("doc")
os.chdir(p)
# try with sample project
build_main(argv=a)

# now use geopandas config
shutil.copy(g.joinpath("source/conf.py"), "source/conf.py")
for f_ in g.joinpath("source/_templates").glob("*.*"):
    shutil.copy(f_, "source/_templates/.")
build_main(argv=a)

# now use about..
shutil.copy(g.joinpath("source/about.md"), "source/.")
p.joinpath("source/about").mkdir()
for f_ in g.joinpath("source/about").glob("*.*"):
    shutil.copy(f_, "source/about/.")
build_main(argv=a)

# now index.rst - hey presto error replicated
shutil.copy(g.joinpath("source/index.rst"), "source/.")
build_main(argv=a)

```

--- a/doc/environment.yml
+++ b/doc/environment.yml
@@ -12,7 +12,7 @@ dependencies:
   - matplotlib
   - mapclassify
   - sphinx
-  - pydata-sphinx-theme=0.9
+  - pydata-sphinx-theme
   - numpydoc
   - ipython
   - pillow

